### PR TITLE
testclient: setChainParams

### PR DESCRIPTION
**Pull request by winsvega**
connects to https://github.com/ethereum/webthree-umbrella/issues/542

### CODE REVIEW

File: libweb3jsonrpc/Test.cpp [hunk 1]
```diff
@@ -32,6 +33,29 @@ using namespace jsonrpc;
 
 Test::Test(eth::Client& _eth): m_eth(_eth) {}
 
+bool Test::test_setChainParams(const Json::Value &param1)
```
Comment (chriseth) [inline libweb3jsonrpc/Test.cpp]: Are these tests run?

Comment (winsvega) [inline libweb3jsonrpc/Test.cpp]: You mean should it belong to test section?
I think so. Normally node is started with either --genesis or --config section

This feature is experimental and for test purpose only.

Comment (chriseth) [inline libweb3jsonrpc/Test.cpp]: Oh sorry, somehow I thought this was a unit test...

Please move the `const` after the type.

File: libweb3jsonrpc/Test.cpp [hunk 2]
```diff
@@ -32,6 +33,29 @@ using namespace jsonrpc;
...
+{
+	try
+	{
+		Json::FastWriter fastWriter;
+		std::string output = fastWriter.write(param1);
+
+		//dev::eth::ChainParams chainParams;
```
Comment (chriseth) [inline libweb3jsonrpc/Test.cpp]: Please remove these.

File: eth/main.cpp [hunk 3]
```diff
@@ -1227,16 +1227,22 @@ int main(int argc, char** argv)
...
-			{
-				h256 contractCodeHash = web3.ethereum()->postState().codeHash(_t.to);
-				if (contractCodeHash == EmptySHA3)
-					return std::make_pair(false, std::string());
-				// TODO: actually figure out the natspec. we'll need the natspec database here though.
-				return std::make_pair(true, std::string());
-			}, [&](Address const& _a) { return ICAP(_a).encoded() + " (" + _a.abridged() + ")"; }
-		) + "\nEnter yes/no/always (always to this address): ", {"yes", "n", "N", "no", "NO", "always"});
+		string r;
+		if (testingMode)
```
Comment (chriseth) [inline eth/main.cpp]: I think the following is both simpler and safer:

```
std::function<bool(TransactionSkeleton const&, bool)> authenticator;
if (testingMode)
  authenticator = [](TransactionSkeleton const&, bool) -> bool { return true; }
else
  authenticator = /* as it was before */
```

File: eth/main.cpp [hunk 4]
```diff
@@ -1222,7 +1222,11 @@ int main(int argc, char** argv)
 
 	AddressHash allowedDestinations;
 
-	auto authenticator = [&](TransactionSkeleton const& _t, bool isProxy) -> bool {
+	std::function<bool(TransactionSkeleton const&, bool)> authenticator;
+	if (testingMode)
+		authenticator = [](TransactionSkeleton const&, bool) -> bool { return true; };
+	else
+		authenticator = [&](TransactionSkeleton const& _t, bool isProxy) -> bool {
 		// "unlockAccount" functionality is done in the AccountHolder.
```
Comment (chriseth) [inline eth/main.cpp]: This still has to be properly indented.

**Comment by chriseth** [comment]
Please merge after indenting the function.

File: eth/main.cpp (hunk 4 above)
Comment (winsvega) [inline eth/main.cpp]: you mean put it inside {} ?

Comment (chriseth) [inline eth/main.cpp]: no, shift the body of the function one tab to the right.

**Pull request merged**